김윤건/week2./베스트 앨범.js

```diff
@@ -0,0 +1,52 @@
+function solution(genres, plays) {
+  const bestAlbum = [];
+  const genresPlayInfo = {};
+  
+  genres.forEach(createGenrePlayInfo.bind(null, genresPlayInfo, plays));
+  
+  const sortedGenres = Object.entries(genresPlayInfo)
+    .sort(sortDescendingOrder)
+    .map(([genre, info]) => {
+      delete info.sum;
+      
+      const sortedSong = Object.entries(info).sort(descendingOrder);
+      
+      return sortedSong;
+    });
+  
+  sortedGenres.forEach(selectBestSong.bind(null, bestAlbum));
+  
+  return bestAlbum
+}
+
+function createGenrePlayInfo (genresPlayInfo, plays, genre, index) {
+  if (genresPlayInfo[genre]) {
+    genresPlayInfo[genre] = {
+      ...genresPlayInfo[genre],
+      [index]: plays[index],
+      sum: genresPlayInfo[genre].sum + plays[index],
+    }
+  } else {
+    genresPlayInfo[genre] = {
+      [index]: plays[index],
+      sum: plays[index],
+    }
+  }
+}
+
+function descendingOrder ([indexA, playsA], [indexB, playsB]) {
+  return playsB - playsA;
+}
+
+function sortDescendingOrder([genreA, infoA], [genreB, infoB]) {
+  return infoB.sum - infoA.sum;
+}
+
+function selectBestSong (bestAlbum, genre) {
+  for (let i = 0; i < genre.length; i++) {
+    if (i >= 2) break;
+      
+    const [index, count] = genre[i];
+    bestAlbum.push(Number(index));
+  }
+}
```

Comment:
```suggestion
function selectBestSong (bestAlbum, genre) {
  for (let i = 0; i < 2; i++) {
    const [index, count] = genre[i];
    bestAlbum.push(Number(index));
  }
}
```

이렇게 까지만 해도 될것 같아서.. ㅎ

Reply:
저도 끝나고 이렇게 수정해둘까 하다가 길이가 2이하일 경우에는 undefined로 나와서 문제가 될 것 같아서 수정해두지 않았었습니다.